core/src/main/java/org/elasticsearch/action/ingest/PutPipelineTransportAction.java

```diff
@@ -60,7 +72,28 @@ protected WritePipelineResponse newResponse() {
 
     @Override
     protected void masterOperation(PutPipelineRequest request, ClusterState state, ActionListener<WritePipelineResponse> listener) throws Exception {
-        pipelineStore.put(clusterService, request, listener);
+        NodesInfoRequest nodesInfoRequest = new NodesInfoRequest();
```

Comment:
same reasoning here. Being correct is more important than the put pipeline API be a bit slower. This is an admin api that won't be invoked many times, so it is the right trade off.